patitasuy/src/app/pages/profile/profile.component.ts

```diff
@@ -3,74 +3,103 @@ import { Post } from 'src/app/models/post.interface';
-        } else if (selected === "favoritos") {
-            this.getFavoritePublications();
-        }
-    }
+  selectPublications(event: any) {
+    const selected = event.detail.value
 
-    getFavoritePublications() {
-        this.postService.getAllFavoritePostsByUser(9).subscribe({
-            next: (res) => {
-                this.publications = res;
-            },
-            error: (error) => {
-                console.log(error);
-            }
-        });
+    if (selected === "mis-publicaciones") {
+      this.getMyPublications();
     }
-
-    getMyPublications() {
-        // TODO: get MY posts instead of favorites
-        this.postService.getAllFavoritePostsByUser(9).subscribe({
-            next: (res) => {
-                this.publications = res;
-            },
-            error: (error) => {
-                console.log(error);
-            }
-        });
+    else if (selected === "favoritos") {
+      this.getFavoritePublications();
     }
+  }
 
-    async openModal() {
-        const modal = await this.modalController.create({
-            component: EditProfileComponent
-        });
-        modal.present();
-        const { data, role } = await modal.onWillDismiss();
 
+  async getFavoritePublications() {
+    let id = 0;
+    this.userService.getCurrentUser().subscribe((userData) => {
+      if (userData?.user_id) {
+        id = userData.user_id;
+      }
+    })
+    this.postService.getAllFavoritePostsByUser(id).subscribe({
+      next: (res) => {
+        this.publications = res;
+      },
+      error: (error) => {
+        console.log(error);
+      }
+    });
+  }
 
-        if (role === 'confirm') {
-            this.name = data.name;
-            this.email = data.email;
+  async getMyPublications() {
+    var id = 0;
+    this.userService.getCurrentUser().subscribe((userData) => {
+      if (userData?.user_id) {
+        id = userData.user_id;
+      }
+      this.postService.getMyPosts(id).subscribe({
+        next: (res) => {
+          this.publications = res;
+        },
+        error: (error) => {
+          console.log(error);
         }
+      });
     }
+    )
+  }
 
   logout() {
-    this.auth.logout();
-    this.router.navigate(['/login']);
+    this.userService.logout();
   }
-}
 
+  async openModal() {
+    const modal = await this.modalController.create({
+      component: EditProfileComponent
+    });
+    modal.present();
+    modal.onDidDismiss().then(async ({ data, role }) => {
+      if (role === 'confirm') {
+        var id = 0;
+        this.userService.getCurrentUser().subscribe((userData) => {
+          if (userData?.user_id) {
+            id = userData.user_id;
+          }
+        })
+        this.userService.setNewUserData(data.name, data.email, id).subscribe({
+          next: (res) => {
+            this.name = res.name;
+            this.email = res.email;
+          },
+          error: (error: any) => {
+            alert(error.error.msg);
+          }
+        })
```

Comment:
This piece of code is not doing what you intended it to do. The second observable MUST be inside the first one, because as of right now, they run at the same time and the first one is not guaranteed to finish before the second one starts. Matter of fact, it will not, it will always send 0 as the id parameter in the setNewUserData method, because that method will be called just as the first observable is subscribed to, so it won't have time to use the ID from the current user and will always use 0

Reply:
you're right, thanks!